moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -292,25 +281,22 @@ void PlanningScene::addCollisionDetector(const collision_detection::CollisionDet
 
   detector->findParent(*this);
 
-  detector->cworld_ = detector->alloc_->allocateWorld(world_);
-  detector->cworld_const_ = detector->cworld_;
+  detector->cenv_ = detector->alloc_->allocateEnv(world_, getRobotModel());
+  detector->cenv_const_ = detector->cenv_;
 
   // Allocate CollisionRobot unless we can use the parent's crobot_.
   // If active_collision_->crobot_ is non-NULL there is local padding and we cannot use the parent's crobot_.
-  if (!detector->parent_ || active_collision_->crobot_)
+  if (!detector->parent_)
```

Comment:
According to the comment here, this happens when there is local padding. Since the equilavent, `active_collision_->cenv_` is always non-null, do we have to assume there is always local padding and this becomes `if (true)`?

Reply:
I am really unsure about all of this parent planning scene. I think the behavior before was:

`cworld_` is always allocated new.  `crobot_` is only allocated if we don't have a parent planning scene or it is the active detector. The unpadded version is allocated if we don't have a parent (although I thought it should only be allocated when the padding is changed...).

With `cenv_` it should be: `cenv_` is always allocated and the unpadded version only when the padding changes?

Reply:
I want to defer to someone else on this, because I never felt like I really figured out the intricacies of these copies and pointers. But I note that reducing this complexity is another big reason why this PR has value, in my opinion.